File: frontend/api/app/core/icons.py
import secrets

# Placeholder icon slugs. Swap/extend with your real icon set later.
# With 24 icons and 3 ordered picks there are 24*23*22 = 12,144 unique combos.
ICON_POOL = [
    "tree", "cat", "apple", "sun", "moon", "star",
    "dog", "fish", "bird", "leaf", "flower", "house",
    "car", "boat", "heart", "cloud", "rain", "snow",
    "fire", "key", "book", "ball", "cake", "bell",
]

ICON_COUNT = 3


def random_icon_set() -> list[str]:
    """Return an ordered list of 3 distinct icon slugs.

    The icon set is effectively the member's password, so draw it from a CSPRNG
    (secrets) rather than the Mersenne-Twister default in `random`.
    """
    pool = list(ICON_POOL)
    return [pool.pop(secrets.randbelow(len(pool))) for _ in range(ICON_COUNT)]


def icons_to_password(icons: list[str]) -> str:
    """The default password is the icon slugs joined: 'tree_cat_apple'."""
    return "_".join(icons)

File: frontend/lib/icons.ts
// Slugs mirror the backend ICON_POOL; keep in sync with core/icons.py.
export const ICON_EMOJI: Record<string, string> = {
  tree: "🌳",
  cat: "🐱",
  apple: "🍎",
  sun: "☀️",
  moon: "🌙",
  star: "⭐",
  dog: "🐶",
  fish: "🐟",
  flower: "🌸",
  house: "🏠",
  car: "🚗",
  boat: "⛵",
  heart: "❤️",
  cloud: "☁️",
  snow: "❄️",
  fire: "🔥",
  book: "📖",
  ball: "⚽",
  cake: "🍰",
  bell: "🔔",
  guitar: "🎸",
  rocket: "🚀",
  crown: "👑",
  gift: "🎁",
  camera: "📷",
  clock: "⏰",
  balloon: "🎈",
  diamond: "💎",
  mushroom: "🍄",
  cactus: "🌵",
  grapes: "🍇",
  lemon: "🍋",
  pizza: "🍕",
};

/**
 * The icons a member can choose from — mirrors ICON_POOL in core/icons.py.
 *
 * Listed explicitly rather than taken from the keys of ICON_EMOJI, because the
 * two answer different questions. This is what may be *chosen*; the map above
 * is what can be *drawn*, and it has to keep every slug that was ever issued so
 * an older account's key still renders as something other than a question mark.
 */
export const ALL_ICONS = [
  "tree",
  "cat",
  "apple",
  "sun",
  "moon",
  "dog",
  "fish",
  "flower",
  "house",
  "car",
  "heart",
  "star",
];

export function emojiFor(slug: string): string {
  return ICON_EMOJI[slug] ?? "❔";
}
